python_modules/dagster/dagster/_core/definitions/asset_condition/scheduling_condition_evaluation_context.py

```diff
@@ -0,0 +1,90 @@
+import dataclasses
+import datetime
+import functools
+from dataclasses import dataclass
+from typing import AbstractSet, Mapping, Optional, Tuple
+
+import pendulum
+
+from dagster._core.asset_graph_view.asset_graph_view import (
+    AssetGraphView,
+    AssetSlice,
+)
+from dagster._core.definitions.asset_condition.asset_condition import (
+    AssetCondition,
+    AssetConditionEvaluation,
+    AssetConditionEvaluationState,
+)
+from dagster._core.definitions.asset_key import AssetKey
+from dagster._core.definitions.asset_subset import ValidAssetSubset
+from dagster._core.definitions.events import AssetKeyPartitionKey
+
+from .asset_condition_evaluation_context import AssetConditionEvaluationContext
+
+
+@dataclass(frozen=True)
```

Comment:
Would love to use DagsterModel here, but was running into Pydantic stuff because we use a `PackableValue` type inside AssetConditionEvaluationState, which causes errors due to its weird type signature.

I'll try to fix that up later in the stack, but for now just using dataclass